builder/googlecompute/block_device.go

```diff
@@ -196,7 +204,7 @@ func (bd *BlockDevice) prepareDiskCreate() []error {
 		errs = append(errs, fmt.Errorf("Scratch volumes cannot have a name specified."))
 	}
 
-	if bd.VolumeSize == 0 {
+	if bd.VolumeSize == 0 && bd.SourceImage == "" {
```

Comment:
I haven't traced how or where this happens but `VolumeSize` appears to be ignored when specifiying a `SourceImage`.

Reply:
Relying on the API is sound indeed. We should probably warn in the docs so that users are aware they have to set the size and that they need to ensure the disk is larger than the base image.

Is the error clearly presented to the user? Or is the excerpt you shared from appears in the verbose logs only?
If it's the latter, we should probably figure out a way to return it clearly, otherwise it looks good to me.

Reply:
Correct me if I'm wrong then, but if we specify a source image, we have to specify the disk size as well, correct? If so this check is not valid anymore

Reply:
> if we specify a source image, we have to specify the disk size as well, correct?

It's still optional I believe. Disk size defaults to the image size if you don't specify it (i.e., 0) which I think is desirable the majority of the time. Here are Google's docs for `DiskSizeGb`:

```
Specifies the size of the disk in base-2 GB. The size must be at least 10 GB. If you specify a sourceImage, which is required for boot disks, the default size is the size of the sourceImage. If you do not specify a sourceImage, the default disk size is 500 GB.
```

> Is the error clearly presented to the user?

I think this was without debug logs. I will double check and get back to you to confirm.